app/src/main/java/com/firebase/uidemo/database/ChatActivity.java

```diff
@@ -93,11 +94,7 @@ public void onComplete(DatabaseError databaseError, DatabaseReference reference)
         });
 
         mMessages = (RecyclerView) findViewById(R.id.messagesList);
-
         mManager = new LinearLayoutManager(this);
-        mManager.setReverseLayout(false);
-
-        mMessages.setHasFixedSize(false);
```

Comment:
There are valid reasons for having this code here, which we could debate here. But these reasons have nothing to do with letting devs intercept change events. In future PRs please keep functional changes separate from code cleanup.